### ISSUE TITLE
Issue with Popen in __init__.py

### ISSUE THREAD

**Issue by xcyan** [cmusatyalab/openface#34]
I tried to run the demo2: "./demos/compare.py images/examples/{lennon*,clapton*}"

But got this error msg:
Traceback (most recent call last):
  File "./demos/compare.py", line 113, in <module>
    d = getRep(img1) - getRep(img2)
  File "./demos/compare.py", line 104, in getRep
    rep = net.forwardImage(alignedFace)
  File "/mnt/brain1/scratch/xcyan/random/openface/demos/../openface/__init__.py", line 81, in forwardImage
    rep = np.array(self.forwardPath(t))
  File "/mnt/brain1/scratch/xcyan/random/openface/demos/../openface/__init__.py", line 75, in forwardPath
    return [float(x) for x in self.p.stdout.readline().strip().split(',')]
ValueError: could not convert string to float: {

I traced the error and found it was line 43 in  __init__.py:
"self.p = Popen(cmd, stdin=PIPE, stdout=PIPE, stderr=PIPE, bufsize=0)"

Basically, this line seems not working (calling torch in python program). Do you have any suggestions? Thanks!

**Comment by bamos** [comment]
Hmm, this still works in the Docker container I built from scratch a few hours ago. For debugging purposes, can you temporarily remove `stdout=PIPE` and send the output from running again? Hopefully we'll get some better information.

**Comment by xcyan** [comment]
Thanks! Let me figure it out.

**Comment by bamos** [comment]
Hi @xcyan - I just pushed some error handling code that will make debugging issues like this easier in the future. No need to remove `stdout=PIPE` manually now. Can you pull the latest from the master branch and try running the demo again?

**Comment by xcyan** [comment]
Thanks! I will update my local files.

**Comment by xcyan** [comment]
}



Error getting result from Torch subprocess.

Line read: {


Exception:

could not convert string to float: {

============

stdout:   imgDim : 96
  cuda : false
  model : "/mnt/brain1/scratch/xcyan/random/openface/demos/../models/openface/nn4.v1.t7"
}


============

stderr:

**Comment by bamos** [comment]
[openface_server.lua](https://github.com/cmusatyalab/openface/blob/master/openface/openface_server.lua)  shouldn't print the options like that. Did you uncomment the line `print(opt)` on line 43?

**Comment by xcyan** [comment]
@bamos No I didn't.

**Comment by xcyan** [comment]
I figured out the issue. Thanks for your suggestions.

**Comment by bamos** [comment]
Good to hear! Did you print the options somewhere else in the server? Or did something else cause this?

**Comment by ajaik91** [comment]
@xcyan  I would like to know how did you solve the error from torch subprocess issue above.

**Comment by xcyan** [comment]
Sorry, it's been a while and I don't remember.

**Comment by eacosta1976** [comment]
I have the Same TORCH Problem running compare.py OpenFace demo....."could not convert string to float" How did you solve it
